src/librustc/query/mod.rs

```diff
@@ -1068,6 +1068,11 @@ rustc_queries! {
             no_force
             desc { "codegen_unit" }
         }
+        query used_generic_parameters(key: DefId) -> BitSet<u32> {
```

Comment:
Would be interesting to run perf with `BitSet` replaced with an `u64` (i.e. parameters after index `64` would have to be assumed used, or I guess they could all share one bit just in case all of them are unused).

Reply:
Fixed in 35b5240.